numpy/core/tests/test_function_base.py

```diff
@@ -29,6 +29,30 @@ def test_dtype(self):
         assert_equal(y.dtype, dtype('int32'))
 
 
+class TestGeomspace(TestCase):
+
+    def test_basic(self):
+        y = geomspace(1, 1e6)
+        assert_(len(y) == 50)
+        y = geomspace(1, 1e6, num=100)
+        assert_(y[-1] == 10 ** 6)
+        y = geomspace(1, 1e6, endpoint=False)
+        assert_(y[-1] < 10 ** 6)
+        y = geomspace(1, 1e6, num=7)
+        assert_array_equal(y, [1, 10, 100, 1e3, 1e4, 1e5, 1e6])
+
+    def test_dtype(self):
+        y = geomspace(1, 1e6, dtype='float32')
+        assert_equal(y.dtype, dtype('float32'))
+        y = geomspace(1, 1e6, dtype='float64')
+        assert_equal(y.dtype, dtype('float64'))
+        y = geomspace(1, 1e6, dtype='int32')
+        assert_equal(y.dtype, dtype('int32'))
+
```

Comment:
At the moment, you are doing whatever `logspace` is doing, and that is fine. I do like the idea of supporting negative numbers though. Either way, you should define and document the behavior, then write a test for it.